pkg/daemon/daemon.go

```diff
@@ -299,6 +306,15 @@ func (dn *Daemon) operatorConfigChangeHandler(old, new interface{}) {
 		dn.disableDrain = newDisableDrain
 		log.Log.Info("Set Disable Drain", "value", dn.disableDrain)
 	}
+
+	if !reflect.DeepEqual(oldCfg.Spec.FeatureGates, newCfg.Spec.FeatureGates) {
+		dn.featureGate.Init(newCfg.Spec.FeatureGates)
+		log.Log.Info("Updated featureGates", "featureGates", dn.featureGate.String())
+	}
+
+	if dn.featureGate.IsEnabled(consts.MellanoxFirmwareResetFeatureGate) {
```

Comment:
you need this in start.go as well so vars.MlxPluginFwReset is initialized on start.

Reply:
And we also need an else clause to disable the feature again